app/models/manageiq/providers/nuage/inventory/collector/target_collection.rb

```diff
@@ -167,6 +168,13 @@ def add_simple_target!(association, ems_ref, options: {})
     target.add_target(:association => association, :manager_ref => {:ems_ref => ems_ref}, :options => options)
   end
 
+  def perform_trivial_vmdb_changes!
+    references_with_options(:cloud_subnet_templates).each do |template|
+      next unless template[:operation] == 'DELETE'
+      manager.cloud_subnets.where('extra_attributes ~* ?', "template_id: #{template[:ems_ref]}\n").destroy_all
```

Comment:
hm, we should probably not be deleting here, we should just fetch the targets. You can add some option to the targets and do not collect the data for them with API.

Reply:
Good point, I did like you suggested and I like it. Also, we'll be able to reuse such approach very easily. Have only two more "Nuage glitches" of this kind to fix, then we're good.